src/pybind/mgr/dashboard/services/nvmeof_client.py

```diff
@@ -24,11 +24,21 @@ class NVMeoFClient(object):
 
         def __init__(self):
             logger.info("Initiating nvmeof gateway connection...")
+            service_name, self.gateway_addr = NvmeofGatewaysConfig.get_service_info()
 
-            self.gateway_addr = list(
-                NvmeofGatewaysConfig.get_gateways_config()["gateways"].values()
-            )[0]["service_url"]
-            self.channel = grpc.insecure_channel("{}".format(self.gateway_addr))
+            server_cert = NvmeofGatewaysConfig.get_server_cert(service_name)
+            client_key = NvmeofGatewaysConfig.get_client_key(service_name)
+            client_cert = NvmeofGatewaysConfig.get_client_cert(service_name)
+
+            if server_cert and client_key and client_cert:
+                credentials = grpc.ssl_channel_credentials(
+                    root_certificates=server_cert,
```

Comment:
BTW, this forces to either share the client certificate across multiple clients, or sign them the others with this certificate. Probably not a big deal, since "client" in the context of Ceph NVMe-oF GW likely represents a small set.